openrouteservice-api-tests/src/test/java/org/heigit/ors/v2/services/routing/ResultTest.java

```diff
@@ -2804,6 +2804,50 @@ public void testPreferGreen() {
                 .statusCode(200);
     }
 
+    @Test
+    public void testPreferShadow() {
+        JSONObject body = new JSONObject();
+        body.put("coordinates", getParameter("coordinatesWalking"));
+
+        given()
+                .header("Accept", "application/json")
+                .header("Content-Type", "application/json")
+                .pathParam("profile", getParameter("footProfile"))
+                .body(body.toString())
+                .when()
+                .post(getEndPointPath() + "/{profile}/json")
+                .then().log().ifValidationFails()
+                .assertThat()
+                .body("any { it.key == 'routes' }", is(true))
+                .body("routes[0].containsKey('summary')", is(true))
+                .body("routes[0].summary.distance", is(2097.2f))
+                .body("routes[0].summary.duration", is(1510.0f))
```

Comment:
In general, we prefer to not test against exact floating point values due to possible rounding errors. You might use `is(closeTo(2097.2, 2))` where the first value is the test-value and the second the error-margin. We typically accept an error of 0.1%-1%.
Maybe you need to insert `.config(RestAssured.config().jsonConfig(jsonConfig().numberReturnType(JsonPathConfig.NumberReturnType.DOUBLE)))` directly after `given()`.

Reply:
I've adapted it as you've described it. I've adapted the `testPreferQuiet` and `testPreferGreen` tests as well, since they are very similar.